src/main/proto_common/common.proto

```diff
@@ -459,30 +465,109 @@ message FailedCancellation {
     string cancellationError = 1;
 }
 
+// All fields optional unless enforced in ProductDescription.
 message Contact {
-    // First name. Mandatory.
     string firstName = 1;
 
-    // Last name. Mandatory.
     string lastName = 2;
 
-    // Title, optional.
-    string title = 3;
+    // Such as Mr. or Mrs.
+    Title title = 3;
```

Comment:
I would leave it as string. This will be filled when parsing response from external API which may return 5-6 different honorifics

Reply:
I have followed platform changes, see https://github.com/Bokun/bokun_platform/blob/develop/app/models/CustomerTitle.java

Reply:
IMO client logic (platform) should decide which title to apply for given string. Plugin will just pass what it got from API. Also if platform will be appended with new titles or with custom title then it will require same change on plugin api. 

Of course it's a minor thing, you can ignore if it not makes much sense.